editor/test/util/diff_test.clj

```diff
@@ -0,0 +1,105 @@
+            :right-lines [""]
+            :edits []}
+           (diff/find-edits "" "")))
+    (is (= {:left-lines ["x"]
+            :right-lines ["x"]
+            :edits [{:type :nop
+                     :left {:begin 0 :end 1}
+                     :right {:begin 0 :end 1}}]}
+           (diff/find-edits "x" "x")))
+    (is (= {:left-lines ["1"
+                         "2"]
+            :right-lines ["1"
+                          "2"]
+            :edits [{:type :nop
+                     :left {:begin 0 :end 2}
+                     :right {:begin 0 :end 2}}]}
+           (diff/find-edits "1\n2" "1\n2"))))
+
+  (testing "Newlines are considered to be part of the prior line."
+    (is (= {:left-lines [""]
+            :right-lines [""
+                          ""]
+            :edits [{:type :insert
+                     :left {:begin 0 :end 0}
+                     :right {:begin 0 :end 1}}]}
+           (diff/find-edits "" "\n")))
+    (is (= {:left-lines ["x"]
+            :right-lines ["x"
+                          ""]
+            :edits [{:type :replace
+                     :left {:begin 0 :end 1}
+                     :right {:begin 0 :end 1}}]}
+           (diff/find-edits "x" "x\n"))))
+
+  (testing "Non-trivial edits."
+    (is (= {:left-lines ["1"
+                         "2"
+                         "3"
+                         ""]
+            :right-lines ["1"
+                          "2"
+                          "3"
+                          "4"
+                          ""]
+            :edits [{:type :nop
+                     :left {:begin 0 :end 3}
+                     :right {:begin 0 :end 3}}
+                    {:type :insert
+                     :left {:begin 3 :end 3}
+                     :right {:begin 3 :end 4}}]}
+           (diff/find-edits "1\n2\n3\n" "1\n2\n3\n4\n")))
+    (is (= {:left-lines ["1"
+                         "2"
+                         "3"
+                         ""]
+            :right-lines ["4"
+                          "1"
+                          "2"
+                          "3"
+                          ""]
+            :edits [{:type :insert
+                     :left {:begin 0 :end 0}
+                     :right {:begin 0 :end 1}}
+                    {:type :nop
+                     :left {:begin 0 :end 3}
+                     :right {:begin 1 :end 4}}]}
+           (diff/find-edits "1\n2\n3\n" "4\n1\n2\n3\n")))
+    (is (= {:left-lines ["1"
+                         "2"
+                         "3"
+                         ""]
+            :right-lines ["2"
+                          "3"
+                          ""]
+            :edits [{:type :delete
+                     :left {:begin 0 :end 1}
+                     :right {:begin 0 :end 0}}
+                    {:type :nop
+                     :left {:begin 1 :end 3}
+                     :right {:begin 0 :end 2}}]}
+           (diff/find-edits "1\n2\n3\n" "2\n3\n")))))
```

Comment:
Moved here from `diff-view-test` with some alterations. We cover all the old stuff, though!